WooCommerce/Classes/Tools/VersionHelpers.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+
+/// Helpers for working with versions (e.g. comparing two version strings)
+///
+final class VersionHelpers {
+    private init() {}
+
+    /// Compares two strings as versions using the same approach as PHP `version_compare`.
+    /// https://www.php.net/manual/en/function.version-compare.php
+    ///
+    /// Returns `orderedAscending` if the lhs version is older than the rhs
+    /// Returns `orderedSame` if the lhs version is the same as the rhs
+    /// Returns `orderedDescending` if the lhs version is newer than the rhs
+    ///
+    static func compare(_ lhs: String, _ rhs: String) -> ComparisonResult {
+
+        /// Replace _ - and + with a .
+        ///
+        func replaceUnderscoreDashAndPlusWithDot(_ string: String) -> String {
+            string.replacingOccurrences(of: "([_\\-+]+)", with: ".", options: .regularExpression)
+        }
+
+        /// Insert a . before and after any non number
+        ///
+        func insertDotsBeforeAndAfterAnyNonNumber(_ string: String) -> String {
+            string.replacingOccurrences(of: "([^0-9.]+)", with: ".$1.", options: .regularExpression)
+        }
+
+        /// Score and compare two string components
+        ///
+        func compareStringComponents(_ lhs: String, _ rhs: String) -> ComparisonResult {
+            /// Score each component
+            let lhsScore = VersionComponentScore(from: lhs)
+            let rhsScore = VersionComponentScore(from: rhs)
+
+            if lhsScore < rhsScore {
+                return .orderedAscending
+            }
+
+            if lhsScore > rhsScore {
+                return .orderedDescending
+            }
+
+            if lhsScore == .number && rhsScore == .number {
+                let lhsAsNumber = NSNumber(value: Int(lhs) ?? 0)
+                let rhsAsNumber = NSNumber(value: Int(rhs) ?? 0)
+
+                let comparisonResult = lhsAsNumber.compare(rhsAsNumber)
+                if comparisonResult != .orderedSame {
+                    return comparisonResult
+                }
+            }
+
+            return .orderedSame
+        }
+
+        /// Process the given string into version components
+        ///
+        func versionComponents(of string: String) -> [String] {
```

Comment:
nit: how about moving these local functions to a private extension, so that it's easier to see the main `compare` code below? totally up to you!

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/5382/commits/2eb5cb0402c9bef48cd7b74fe3d81bbe031f3c44